packages/yarnpkg-cli/sources/main.ts

```diff
@@ -90,6 +90,7 @@ export async function main({binaryVersion, pluginConfiguration}: {binaryVersion:
         const iShouldBeHere = realpathSync(cwd);
 
         if (iAmHere !== iShouldBeHere) {
+          process.argv.splice(process.argv.findIndex((arg) => arg === `--cwd`), 2);
```

Comment:
Not a fan of the slice. In particular, someone running `--cwd=foo` will hit weird problems 🤔

Can we just add a parameter to run (`enableCwd`, defaults to true) that we would set to `false` in the recursive call?

Reply:
Totally forgot that `--cwd=foo` is also valid.
There's a problem with the `enableCwd` approach: It still only gets checked on the else branch, so the issue wouldn't be solved. I don't see a way for it to be passed to the if branch without changing a lot of things.

Reply:
And an environment variable, similar to what we do with the `YARN_PATH`? For example we could set `YARN_IGNORE_CWD`, and delete it in subprocesses.

A repro test would also help to better understand under which circumstances the bug is triggered

Reply:
The bug is triggered every time a relative `--cwd` (the person who opened #1134 says that absolute paths work, I haven't tested it myself) is provided:
```js repro
const {promises: {mkdir}} = require(`fs`);

await mkdir(`./subdir`);

await yarn(`./subdir`, `init`);

await expect(yarn(`--cwd`, `./subdir`)).rejects.not.toThrow(`ENOENT: no such file or directory, lstat`);
```
I'll try the environment variable approach.

Reply:
The environment variable approach works.😀